### PR TITLE
Moved over the Amazon pipeline

### PR DESCRIPTION
**Pull request by tomerk**
issue #165 
@etrain

### CODE REVIEW

**Comment by tomerk** [comment]
Okay @etrain how's this look now?

File: src/main/scala/pipelines/text/AmazonReviewsPipeline.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import pipelines.Logging
+import scopt.OptionParser
+import workflow.Optimizer
+
+object AmazonReviewsPipeline extends Logging {
+  val appName = "AmazonReviewsPipeline"
+
+  def run(sc: SparkContext, conf: AmazonReviewsConfig) {
+
+    logInfo("PIPELINE TIMING: Started training the classifier")
```
Comment (etrain) [inline src/main/scala/pipelines/text/AmazonReviewsPipeline.scala]: This probably doesn't need to be there (nor do its counterparts at the end.)

File: src/main/scala/pipelines/text/NewsgroupsPipeline.scala [hunk 2]
```diff
@@ -21,7 +21,8 @@ object NewsgroupsPipeline extends Logging {
 
     // Build the classifier estimator
     logInfo("Training classifier")
-    val predictorPipeline = Trim andThen LowerCase() andThen
+    val predictorPipeline = Trim andThen
+        LowerCase() andThen
```
Comment (etrain) [inline src/main/scala/pipelines/text/NewsgroupsPipeline.scala]: The `()` need to be here?

Comment (tomerk) [inline src/main/scala/pipelines/text/NewsgroupsPipeline.scala]: Yup, see my previous comment.

**Comment by etrain** [comment]
So I think the main thing is going to be moving over a reasonable cache-oblivious Logistic Regression port, but other than that, the extraneous log statements, and the little formatting (possibly non-) issues, I think this is good to go.

File: src/main/scala/pipelines/text/AmazonReviewsPipeline.scala (hunk 1 above)
Comment (tomerk) [inline src/main/scala/pipelines/text/AmazonReviewsPipeline.scala]: Aww you don't like my timing logging?

File: src/main/scala/pipelines/text/NewsgroupsPipeline.scala (hunk 2 above)
Comment (etrain) [inline src/main/scala/pipelines/text/NewsgroupsPipeline.scala]: Ah, right. Stupid scala.

**Comment by tomerk** [comment]
Okay, how does this look now? (using copied code instead of reflection)

**Comment by tomerk** [comment]
@etrain

**Comment by tomerk** [comment]
Okay @etrain how about now? I didn't figure out a way to override the numFeatures directly via the constructor params. This does compile because the only things it uses from MLlib are marked DeveloperAPI. I haven't tested this code though, so not 100% positive yet that it runs (and we should have a unit test for the logistic regression too)

**Comment by etrain** [comment]
Well - lets kill two birds with one stone and use the unit test to verify? An assert on the number of rdds in cache would probably be good too in that one.

**Comment by tomerk** [comment]
Okay @etrain we have the unit test to verify that the logistic regression works (ported over from mllib). Not sure how to test the number of rdds in cache though...

**Comment by tomerk** [comment]
Also @shivaram

**Comment by etrain** [comment]
Re: number of RDDs in cache - see SparkContext.getRDDStorageInfo

**Comment by tomerk** [comment]
Hmm so it looks like the "addIntercept" that the tests use also forces a cache to occur, but it's not a mandatory parameter for the node itself.

**Comment by etrain** [comment]
So update the unit test?

**Comment by tomerk** [comment]
Okay, I did what we talked about and removed the addIntercept param.

**Pull request merged**